### ISSUE TITLE
Luaver kill the whole batch process on error

### ISSUE THREAD

**Issue by umireon** [DhavalKapil/luaver#29]
The following script will not work as expected:

```bash
. ~/.luaver/luaver
if luaver install-luarocks install 5.3.1; then
    echo ok
else
    echo Something wrong
fi
```

The result was:

```bash
$ sh test.sh
No lua version set

$
```

while printing `Something wrong` is the expected behavior.

This is caused by [luaver#L34](https://github.com/DhavalKapil/luaver/blob/be7d745ea89d0f90cefecb178f2740a2484cbae9/luaver#L34), which kills not only the `luaver` function itself but also the whole batch process itself.
This behavior makes difficult to use luaver on the CI.

Since there is no exception handling mechanism on shells (`kill -INT $$` is appearently not working),  I suggest each command has explicit error handling using `&&` and `if`.

**Comment by DhavalKapil** [comment]
Can you elaborate on the issue. I see it is a problem but exactly how is it affecting the CI?

**Comment by umireon** [comment]
(I am sorry but I cannot fully understand why `kill -INT $$` works; I cannot find any spec to describe what is happen when I shoot myself in the foot with shell by `SIGINT`)

**tl;dr** `luaver` behaves like `exit` on error and this is hard to understand since no one expect that.

Of course, the current luaver is **working** when there are no errors.
The problem is in **debugging**, the behavior of `kill -INT $$` makes debugging very hard.

For example, [travis_retry](https://docs.travis-ci.com/user/common-build-problems/#travis_retry) won't work as expected.
In CI sessions, retrying is essential for stable builds because download failure is one of the common problems.
The following code is a typical pattern to realize the retrying.

```yaml
install: for i in 1 2 3; do luaver install 5.3.2; done
# essentially equivalent to `travis_retry luaver install 5.3.2`
```

However, this will never work as expected because luaver kills the ongoing shell process on error, which cause the loop to stop.
If the user is suficiently wise, they will conceive that luaver does something intrusive to handle errors.
The current error handling proccess is `exit`ting the current shell without writing `exit`.
More concisely, `luaver install 5.3.2` is `exit 1` under certain conditions.

**Comment by umireon** [comment]
The behavior of `kill -INT $$` in interactive sessions is different among shells and exit code cannot be controlled.
Even worse, `zsh` masks the error.

## bash

```bash
$ kill -INT $$

$ echo $?
1
```

## zsh

```bash
% kill -INT $$
% echo $?
0
```

## dash

```
$ kill -INT $$

$ echo $?
130
```

## ksh

```
$ kill -INT $$
$ echo $?
258
```

**Comment by DhavalKapil** [comment]
I get your point. This is indeed a minor yet important bug. An alternate way needs to be found out. I'm ignoring this for the present release though.
